apache/superset
# chore: Notify user of custom label colors and related Dashboard color scheme

### SUMMARY

This PR: 

- Adds the ability to notify the user about custom label colors used in a Dashboard
- Adds the ability to inform the user when a chart in Explore is having a dashboard context, thus the color scheme will be inherited from the dashboard.
- Adds some test cases for the ColorSchemeControl which had none before
- Fixes #17318

### SUPERSET-UI PR
Requires https://github.com/apache-superset/superset-ui/pull/1467 to be merged to function fully

### AFTER DASHBOARD

https://user-images.githubusercontent.com/60598000/141459740-af81102e-c468-4fd7-855d-535c5c280804.mp4

### AFTER EXPLORE

https://user-images.githubusercontent.com/60598000/141459760-df5ed5b2-5888-4554-bb79-b116f7b1e5e3.mp4

### TESTING INSTRUCTIONS
1. Open a Dashboard
2. Set a color scheme
3. Add custom label colors
4. Make sure the color scheme alerts you of custom label colors in use
4. Visit a chart
5. Make sure the color scheme cannot be changed in Explore

### ADDITIONAL INFORMATION


- [x] Has associated issue: #17318
